<launch>
  <!-- rosbag play -->
  <arg name="bag"/>
  <param name="use_sim_time" value="true"/>
  <node pkg="rosbag" type="play" name="rosbag_play" args="--clock $(arg bag)" required="true"/>
  
  <node pkg="velodyne_pointcloud" type="cloud_node" name="cloud_node">
    <param name="calibration" value="$(find velodyne_pointcloud)/params/VLP16db.yaml"/>
    <param name="min_range" value="0.4"/>
    <param name="max_range" value="50.0"/>
  </node>
  
  <!-- Velodyne VLP-16 Driver
       <node pkg="nodelet" type="nodelet" name="velodyne_nodelet_manager" args="manager" output="screen"/>
       
       <node pkg="nodelet" type="nodelet" name="driver_nodelet" args="load velodyne_driver/DriverNodelet velodyne_nodelet_manager" output="screen">
       <param name="model" value="VLP16"/>
       <param name="pcap" value=""/>
       <param name="frame_id" value="velodyne"/>
       <param name="rpm" value="600.0"/>
       </node>
       
       <node pkg="nodelet" type="nodelet" name="cloud_nodelet" args="load velodyne_pointcloud/CloudNodelet velodyne_nodelet_manager" output="screen">
       <param name="calibration" value="$(find velodyne_pointcloud)/params/VLP16db.yaml"/>
       <param name="min_range" value="0.4"/>
       <param name="max_range" value="100.0"/>
       </node>
  -->
  
  <!-- FLOBOT 3D Object Detector -->
  <node pkg="object3d_detector" type="object3d_detector" name="object3d_detector" output="screen" respawn="false">
    <param name="print_fps" type="bool" value="true"/>
    <param name="model_file_name" type="string" value="$(find object3d_detector)/model/pedestrian.model"/>
    <param name="range_file_name" type="string" value="$(find object3d_detector)/model/pedestrian.range"/>
    <param name="human_size_limit" type="bool" value="true"/>
  </node>
  
  <!-- ROS Visualization -->
  <param name="robot_description" command="cat $(find flobot_tracker)/urdf/pioneer3at.urdf" />
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find object3d_detector)/launch/object3d_detector.rviz"/>
</launch>
